plugins/EventGhost/__init__.py

```diff
@@ -608,20 +644,75 @@ class text:
         text1 = "Event string to fire:"
         text2 = "Delay the firing of the event:"
         text3 = "seconds. (0 = fire immediately)"
+        text4 = "Add event to event queue:"
+        text5 = "Return eg.event to original event:"
 
-    def __call__(self, eventString, waitTime=0):
+    def __call__(
+        self,
+        eventString,
+        waitTime=0,
+        queueEvent=True,
+        restoreEvent=False
+    ):
         eventString = eg.ParseString(eventString)
         if not waitTime:
-            eg.TriggerEvent(eventString)
+            if queueEvent:
+                eg.TriggerEvent(eventString)
+            else:
+                event = eg.EventGhostEvent(suffix=eventString)
+                if restoreEvent:
+                    old_event_string = eg.eventString
+                    old_event = eg.event
+                    event.Execute()
+                    eg.event = old_event
+                    eg.eventString = old_event_string
+                else:
+                    event.Execute()
         else:
             eg.scheduler.AddShortTask(waitTime, eg.TriggerEvent, eventString)
 
-    def Configure(self, eventString="", waitTime=0):
+    def Configure(
+        self,
+        eventString="",
+        waitTime=0,
+        queueEvent=False,
+        restoreEvent=False
+    ):
         panel = eg.ConfigPanel()
         text = self.text
 
         eventStringCtrl = panel.TextCtrl(eventString, size=(250, -1))
         waitTimeCtrl = panel.SpinNumCtrl(waitTime, integerWidth=5)
+        queueEventCtrl = wx.CheckBox(panel, -1, '')
+        restoreEventCtrl = wx.CheckBox(panel, -1, '')
+
+        queueEventCtrl.SetValue(queueEvent)
+        restoreEventCtrl.SetValue(restoreEvent)
+        queueEventText = panel.StaticText(text.text4)
+        restoreEventText = panel.StaticText(text.text5)
+
+        def on_spin(evt):
+            def check_spin():
+                value = bool(waitTimeCtrl.GetValue())
+                queueEventText.Show(not value)
```

Comment:
I would preferr to disable the options instead of hiding them.

Reply:
ok No problem there either.